src/common/DOMWorld.ts

```diff
@@ -512,9 +512,12 @@ export class DOMWorld {
     const bind = async (name: string) => {
       const expression = helper.pageBindingInitString('internal', name);
       try {
+        // TODO: In theory, it would be enough to call this just once
```

Comment:
Would the binding also survive page navigations if we only call it once?

Reply:
The binding is added to `m_activeBindings` in both cases and stays there until the binding is removed by calling the `removeBinding` cdp command. And whenever a new `ExecutionContext` is created, it automatically restores the bindings. This works except for the case when the binding is only exposed to a single execution context identified by an id because it will not be able to restore it.

Adding the binding by name will therefore restore it after navigations. I also added a test for that case to make sure that we don't regress.

Reply:
Ah, I see. It sounds like addressing contexts by name instead of id is quite appropriate for this use case then. Thanks!